backend/inspirehep/tools/views.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from io import BytesIO, TextIOWrapper
+
+from flask import Blueprint, abort, current_app, jsonify, request
+
+from inspirehep.files.api import current_s3_instance
+from inspirehep.records.utils import hash_data
+from inspirehep.tools.utils import (
+    find_references,
+    get_filename,
+    get_mimetype,
+    get_references,
+)
+
+blueprint = Blueprint("inspirehep_tools", __name__, url_prefix="/tools")
+
+ALLOWED_EXTENSIONS = {"tex"}
+
+
+def allowed_file(filename):
+    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@blueprint.route("/bibgen", methods=["POST"])
+def generate_bibliography():
+
+    requested_format = request.args.get("format")
+
+    bytes_file = request.files["file"]
+
+    if not allowed_file(bytes_file.filename):
+        abort(400)
+
+    text_file = TextIOWrapper(bytes_file, errors="ignore")
+
+    reference_names = get_references(text_file)
+    references, errors = find_references(reference_names, requested_format)
+
+    file_data = "\n\n".join(references).encode("utf-8")
+    key = hash_data(file_data)
+    file_data = BytesIO(file_data)
+    filename = get_filename(bytes_file.filename, requested_format)
+    mime_type = get_mimetype(requested_format)
+    bucket = current_app.config.get("S3_BIBGEN_BUCKET")
+
+    current_s3_instance.upload_file(
+        file_data, key, filename, mime_type, current_app.config["S3_FILE_ACL"], bucket
+    )
```

Comment:
what's the plan for garbage collecting these temporary files? we ignore the issue? we add an S3 lifecycle to the bucket to expire objects after some time? https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html

Reply:
We could set an expiration policy of one day for example. I'll have a look at it.